task_processing/plugins/mesos/execution_framework.py

```diff
@@ -178,19 +181,22 @@ def unblacklist_slave(self, agent_id, timeout):
             self.blacklisted_slaves = \
                 self.blacklisted_slaves.remove(agent_id)
 
-    def enqueue_task(self, task_config):
+    def enqueue_task(self, task_config, task_id):
+        mesos_task_id = ':'.join(task_id)
```

Comment:
I am a bit confused about the need for mesos_task_id and task_id. Why do we need mesos_task_id? IMO, it is complicating the code a lot.